src/maputils.erl

```diff
@@ -35,6 +36,18 @@ deep_get([Key|Keys], Map) ->
 deep_get([], Value) ->
     Value.
 
+deep_put([Key|Keys], Value, Map) when is_map(Map) ->
+    case maps:is_key(Key, Map) of
+        true ->
+            Map#{Key := deep_put(Keys, Value, maps:get(Key, Map))};
+        false ->
+            Map#{Key => deep_put(Keys, Value, #{})}
+    end;
+deep_put([], Value, _) ->
+    Value;
+deep_put(_, _, Map) ->
```

Comment:
This case will also happen if the first argument isn't a list. I would rather crash on a function clause on that, then exit and blame the Map. Alternatively, check if the first arg isn't a list and then exit on that.